File: articles/calculus-for-systems-modeling/articles/financial-dynamics-and-continuous-compounding/fortran/financial_dynamics_compounding.f90
program financial_dynamics_compounding
  implicit none
  real(8) :: fv, pv
  fv = 1000.0d0 * exp(0.05d0 * 30.0d0)
  pv = 5000.0d0 * exp(-0.05d0 * 30.0d0)
  print '(A)', 'scenario final_value present_value'
  print '(A,1X,F12.6,1X,F12.6)', 'continuous_compounding_case', fv, 1000.0d0
  print '(A,1X,F12.6,1X,F12.6)', 'discounted_future_value', 5000.0d0, pv
end program

File: articles/calculus-for-systems-modeling/articles/financial-dynamics-and-continuous-compounding/c/financial_dynamics_compounding.c
#include <math.h>
#include <stdio.h>
#include <stdlib.h>

double continuous_future_value(double v0, double r, double t){
  return v0 * exp(r * t);
}

double continuous_present_value(double fv, double r, double t){
  return fv * exp(-r * t);
}

int main(void){
  printf("scenario_name,model_type,final_value,present_value,warning\n");
  printf("continuous_compounding_case,future_value,%.6f,1000.000000,continuous_compounding\n", continuous_future_value(1000.0,0.05,30.0));
  printf("discounted_future_value,present_value,5000.000000,%.6f,discounting\n", continuous_present_value(5000.0,0.05,30.0));
  return EXIT_SUCCESS;
}
